keymaster/OpteeKeymaster.cpp

```diff
@@ -84,26 +76,35 @@ int OpteeKeymaster::Initialize() {
     Configure(req, &rsp);
 
     if (rsp.error != KM_ERROR_OK) {
-        ALOGE("Failed to configure keymaster %d", rsp.error);
+        LOG(FATAL) << "Failed to configure keymaster " << rsp.error;
         return -1;
     }
 
+    // Set the vendor patchlevel to value retrieved from system property (which
+    // requires SELinux permission).
+    ConfigureVendorPatchlevelRequest vendor_req(message_version());
+    vendor_req.vendor_patchlevel = GetVendorPatchlevel();
+    ConfigureVendorPatchlevelResponse vendor_rsp = ConfigureVendorPatchlevel(vendor_req);
+    if (vendor_rsp.error != KM_ERROR_OK) {
+        LOG(ERROR) << "Failed to configure keymaster vendor patchlevel: " << vendor_rsp.error;
+        // Don't fail if this message isn't understood.
```

Comment:
Could you explain more in detail (as a comment in the code), why this is OK, i.e., to not return any error if this happens.

Reply:
No idea, it's copy-paste from trusty ...

In our fork we let the TA handle this request:
https://gitlab.baylibre.com/baylibre/ti/android/aosp/external/kmgk/-/blob/ti-android-14/keymaster/ta/keystore_ta.c?ref_type=heads#L1819

The function was stubbed which means we return `KM_ERROR_OK` but we do nothing.

So I think either we do this kind of logic or we remove this code ?